WordPress/gutenberg
# Block editor: localize search suggestions' post type in LinkControl component (#21377)

## What?
Show localized name for post type for suggestions in `LinkComponent`

## Why?
Currently, suggestions are always displayed with their post type in English.

## How?
- Add a new media search handler to standardize the response shape for all search request in the editor
- Add a new REST field `label` in the search-result schema. This field contains the localized post type name of each item in the response.
- Update `getVisualTypeName` function to use the new `label` field when displaying suggestions in `LinkControl`

## Testing Instructions
In the WordPress admin, using a custom locale (other than English)
1. Create a few posts / pages / attachment
2. In a new post, insert a Button block
3. Select the link option in the block toolbar and search for suggestion

In the results list, the post type next to each item should be displayed in the correct locale.

## Screenshots or screencast
![patch-21377](https://github.com/WordPress/gutenberg/assets/158572/97ee1479-b728-43ab-b9bb-afdac9689dcb)

## Other notes
If this solution is approved, we'll need to open a trac ticket in Core to add the new media search handler and add the new `label` field. These changes in the REST are currently polyfilled in this PR.
